# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: docshell/base/CanonicalBrowsingContext.h
# Commit: bb05fe8fa742
# Full Hash: bb05fe8fa742e7e470509f52a957d3b82ced2acb
# Author: Olli Pettay <Olli.Pettay@helsinki.fi>
# Date: 2021-06-04 10:21:11
# Description:
#   Bug 1713365, don't try to bfcache if the session history entry already has a frameloader, r=peterv
#   
#   The crash happens currently when there are more than one loading session history entries at the same time and when
#   the latter load then accesses current active entry, it has already the bfcached frameloader.
#   
# ==============================================================================

diff -r 52ed1b7f1fb0 -r bb05fe8fa742 docshell/base/CanonicalBrowsingContext.h
--- a/docshell/base/CanonicalBrowsingContext.h	Thu Jun 03 20:31:22 2021 +0000
+++ b/docshell/base/CanonicalBrowsingContext.h	Thu Jun 03 20:34:24 2021 +0000
@@ -11,6 +11,7 @@
 #include "mozilla/dom/MediaControlKeySource.h"
 #include "mozilla/dom/BrowsingContextWebProgress.h"
 #include "mozilla/dom/Promise.h"
+#include "mozilla/dom/SessionHistoryEntry.h"
 #include "mozilla/dom/SessionStoreRestoreData.h"
 #include "mozilla/dom/SessionStoreUtils.h"
 #include "mozilla/dom/ipc/IdType.h"
@@ -47,7 +48,6 @@
 struct LoadURIOptions;
 class MediaController;
 struct LoadingSessionHistoryInfo;
-class SessionHistoryEntry;
 class SSCacheCopy;
 class WindowGlobalParent;
 
@@ -58,6 +58,7 @@
   bool mReplaceBrowsingContext = false;
   uint64_t mSpecificGroupId = 0;
   bool mTryUseBFCache = false;
+  RefPtr<SessionHistoryEntry> mActiveSessionHistoryEntry;
 };
 
 // CanonicalBrowsingContext is a BrowsingContext living in the parent